staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go

```diff
@@ -135,7 +137,10 @@ func AsPartialObjectMetadata(m metav1.Object) *metav1.PartialObjectMetadata {
 				OwnerReferences:            m.GetOwnerReferences(),
 				Finalizers:                 m.GetFinalizers(),
 				ClusterName:                m.GetClusterName(),
-				ManagedFields:              m.GetManagedFields(),
+				// Explicitely set ManagedFields to nil
+				// since it's big and not necessary for
+				// metadata-only clients.
```

Comment:
// TODO: we may need to pass this through if requested by the user. But this isn't particularly useful without the entire object, so maybe not.

Reply:
Yeah, I thought about that, but "not passing" is probably a reasonable default, so we can provide a mechanism later.

Reply:
That's my thought too, I just think we should probably say so in the comment.

Reply:
I'll update once we're done with the first round of tests.